### DISCUSSION TITLE
Considering adding new imported MissionControl data (XImportMissionControl) to the database not only into memory

### DISCUSSION THREAD

**Discussion by ziggie1984** [lightningnetwork/lnd#8840] [General]
Related to the SoB project (https://github.com/ziggie1984/Distributed-Mission-Control-for-LND) we found out that it would make sense to persist imported Mission Control data across restarts. The goal of the mentioned project is to help light-clients e.g. non-custodial wallets like `blixt` to increase their payment success rate hence offering a better user-experience but also reducing the risk of stuck htlcs. 

In the current RPC call we only change the data in memory: https://github.com/lightningnetwork/lnd/blob/83b1c22e794e09e3f4ced144695dbe6f6e9165d0/routing/missioncontrol.go#L363-L384

Is there an obvious pitfall I am not considering when introducing the possiblity to also update the Mission Data store in addition to the in-memory state ?

Otherwise we would like to introduce this option to LND in the course of the SoB project.

Thank you in advance.

**Comment by saubyk** [comment]
I don't really see any harm in persisting the imported MC data to storage.

cc: @bitromortac @ellemouton for their thoughts on this

**Reply by bitromortac** [reply]
I think it is useful, it may make sense to study any performance changes (depending on the number of entries added). It may also be needed to add some sanity checks for the data (timestamps should not be in the future for example, etc).

**Reply by ellemouton** [reply]
I think we should maybe add a flag to explicitly allow it (or disallow...) cause i think the OG reason was to be able to test various updates to see how they would affect payment success without having those tests now be persisted permanently incase they degraded the success rate
